File: packages/core-native/ios/PhotosCoreModule.swift
import ExpoModulesCore

// Thin Expo Module over the UniFFI-generated Swift bindings (`ios/generated`,
// produced by `bun run --filter @photos/core-native build:rust`). Keeps the JS
// surface identical to the WASM build: JSON in, JSON out.
public class PhotosCoreModule: Module {
  private let core = Core()

  public func definition() -> ModuleDefinition {
    Name("PhotosCore")

    Function("version") { () -> String in
      version()
    }

    Function("dispatch") { (eventJson: String) throws in
      try core.dispatch(eventJson: eventJson)
    }

    Function("stateJson") { () -> String in
      core.stateJson()
    }
  }
}

File: packages/core-native/android/src/main/java/app/photos/core/PhotosCoreModule.kt
package app.photos.core

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import uniffi.photos_core_ffi.Core
import uniffi.photos_core_ffi.version

// Thin Expo Module over the UniFFI-generated Kotlin bindings
// (src/main/java/uniffi, produced by `bun run --filter @photos/core-native build:rust`).
class PhotosCoreModule : Module() {
  private val core = Core()

  override fun definition() = ModuleDefinition {
    Name("PhotosCore")

    Function("version") { version() }

    Function("dispatch") { eventJson: String -> core.dispatch(eventJson) }

    Function("stateJson") { core.stateJson() }
  }
}
